save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/docker/DockerAgentRunner.kt

```diff
@@ -67,22 +69,28 @@ class DockerAgentRunner(
         }
     }
 
-    override fun stop(executionId: String) {
+    override fun stop(executionId: Long) {
         val runningContainersForExecution = dockerClient.listContainersCmd().withStatusFilter(listOf("running")).exec()
             .filter { container -> container.names.any { it.contains("-$executionId-") } }
         runningContainersForExecution.map { it.id }.forEach { agentId ->
             dockerClient.stopContainerCmd(agentId).exec()
         }
     }
 
-    override fun stopByAgentId(agentId: String) {
+    override fun stopByAgentId(agentId: String): Boolean {
         logger.info("Stopping agent with id=$agentId")
         val state = dockerClient.inspectContainerCmd(agentId).exec().state
-        if (state.running == true) {
-            dockerClient.stopContainerCmd(agentId).exec()
+        return if (state.running == true) {
```

Comment:
if (state.running) ?
Or it can be null?

If null - then we should log it properly:
`"Agent with id=$agentId was requested to be stopped, but it actual state=$state"`

Reply:
Yep, it's nullable, because it's marked as `@CheckForNull` in Java code. However, [Docker API docs](https://docs.docker.com/engine/api/v1.41/#operation/ContainerInspect) don't specify when it can be absent. They suggest, however, to use `Status == running` instead of `running` boolean field.